### DISCUSSION TITLE
How to create a content item from a workflow property as JSON?

### DISCUSSION THREAD

**Discussion by willnationsdev** [OrchardCMS/OrchardCore#8888] [Q&A]
Within a Script task, I want to build a JSON structure based on values inside Form. I will then assign this value to a property A.

In a later Script task, I grab this property as a JSON object, add some more data to it, and resave it as a different property B.

Finally, in a CreateContentTask, I want to inject the entirety of the JSON structure contained in property B as my "expression".

I cannot figure out how I'm supposed to do this.

- When doing `setProperty("B", { "x": "y" })`, `{{ Workflow.Properties.B }}` does not appear to properly render a JObject, so it must be stored as something else. It looks fine in the "State" report of the workflow instance, but when you try to get the property in the Liquid expression for CreateContentTask, you just instead get a report on how many keys there are and what their names are. The Liquid expression then fails to properly parse this.
- `{{ Workflow.Properties.B | json }}` doesn't appear to work either if you just want to inject the JSON string as the expression. I get a fault saying that some parameter 's' cannot be null (no idea where it's coming from).
- If you attempt to call `setProperty("B", JSON.stringify({ ... })`, the corresponding `{{ Workflow.Properties.B | jsonparse }}` fails to properly parse the string, saying it's running into unrecognizable backslashes (is it escaping the quotes?). And attempting to remove the backslashes with replace filters in Liquid doesn't seem to help.
- Furthermore, if you try to call `JSON.stringify()` on any value based on Properties or Output, it triggers an infinite loop that prevents the workflow from ever terminating, and *eventually* the process itself will crash because of a stack overflow exception.

Any guidance would be appreciated.

**Comment by Skrypt** [comment]
Let me create a Workflow with a form quick to remember the details. I will link the recipe here if I can.

**Reply by syampolsky** [reply]
Any resolution to this?

**Reply by Skrypt** [reply]
We need to repro first to see where the encoding fails if any.
